sashimi/gui/main_gui.py

```diff
@@ -143,6 +143,32 @@ def check_end_experiment(self):
             self.st.saver.saver_stopped_signal.clear()
             self.toolbar.experiment_toggle_btn.flip_icon(False)
 
+        # check if experiment started or ended and update gui enabling
+        if self.st.is_exp_started():
+            self.disable_gui()
+        elif self.st.is_exp_ended():
+            self.enable_gui()
+
+    def disable_gui(self):
+        """Disable all the gui elements during the experiment"""
+        self.menuBar().setEnabled(False)
+        self.wid_laser.setEnabled(False)
+        self.wid_status.setEnabled(False)
+        self.wid_scan.setEnabled(False)
+        self.wid_camera.setEnabled(False)
+        self.wid_save_options.setEnabled(False)
+        self.wid_display.auto_contrast_chk.setEnabled(False)
+
+    def enable_gui(self):
+        """Enables all the gui elements after the end of the experiment"""
+        self.menuBar().setEnabled(True)
+        self.wid_laser.setEnabled(True)
+        self.wid_status.setEnabled(True)
+        self.wid_scan.setEnabled(True)
+        self.wid_camera.setEnabled(True)
+        self.wid_save_options.setEnabled(True)
+        self.wid_display.auto_contrast_chk.setEnabled(True)
+
```

Comment:
Here the main gui checks if the experiment started or ended and calls the respective function to enable/disable the gui